sos/report/plugins/charmed_mysql_k8s.py

```diff
@@ -0,0 +1,188 @@
+        "MYSQL_SHELL_LOGS": "/var/log/mysqlsh",
+    }
+
+    option_list = [
+        PluginOpt(
+            "dbuser", default="mysql", val_type=str,
+            desc="Username for database dump collection"
+        ),
+        PluginOpt(
+            "dbpass", default="", val_type=str,
+            desc="Password for database dump collection",
+        ),
+        PluginOpt(
+            "dumpdbs", default=False, val_type=bool,
+            desc="Get name of all databases"
+        ),
+        PluginOpt(
+            "logs_since", default="48h", val_type=str,
+            desc="How far back to fetch logs with kubectl --since"
+        ),
+    ]
+
+    kube_cmd = "kubectl"
+    selector = "app.kubernetes.io/name=mysql-k8s"
+
+    def _determine_namespaces(self):
+        namespaces = self.exec_cmd(
+            f"{self.kube_cmd} get pods -A -l {self.selector} "
+            "-o jsonpath='{.items[*].metadata.namespace}'"
+        )
+        if namespaces['status'] == 0:
+            return list(set(namespaces['output'].strip().split()))
+        return []
+
+    def _get_pod_names(self, namespace):
+        pods = self.exec_cmd(
+            f"{self.kube_cmd} -n {namespace} get pods -l {self.selector} "
+            "-o jsonpath='{.items[*].metadata.name}'"
+        )
+        if pods['status'] == 0:
+            return pods['output'].strip().split()
+        return []
+
+    def _collect_per_namespace(self, namespace, logs_since):
+        kube_cmd = f"{self.kube_cmd} -n {namespace}"
+
+        # Describe the resources
+        self.add_cmd_output([
+            f"{kube_cmd} get all -l {self.selector} -o wide",
+            f"{kube_cmd} describe pods -l {self.selector}",
+        ])
+
+        # Get pod logs
+        self.add_cmd_output(
+            f"{kube_cmd} logs -l {self.selector} --since={logs_since} "
+            "--all-containers=true --prefix --all-pods"
+        )
+
+        mysql_cont = "mysql"
+        pods = self._get_pod_names(namespace)
+
+        # Get the config and logs from each pod
+        dump_files_path = self.get_cmd_output_path()
+        for name, path in self.mysql_conf_paths.items():
+            for pod in pods:
+                os.makedirs(f"{dump_files_path}/{pod}/{name}", exist_ok=True)
+                copy_cmd = (
+                    f"{kube_cmd} cp -c {mysql_cont} {pod}:{path} "
+                    f"{dump_files_path}/{pod}/{name}"
+                )
+                self.exec_cmd(copy_cmd)
+
+        self.add_forbidden_path([
+            f"{dump_files_path}/*/MYSQL_CONF/*.pem",
+            f"{dump_files_path}/*/MYSQL_CONF/*.key",
+        ])
+        for pod in pods:
+            self.add_copy_spec(f"{dump_files_path}/{pod}/*")
+
+        # If dumpdbs is set, then get all databases
+        if self.get_option("dumpdbs"):
+            db_user = self.get_option("dbuser")
+            db_pass = self.get_option("dbpass")
+            pass_file = "/tmp/mysql_pwd"
```

Comment:
I appreciate this is expected to run in relatively constrained k8s environments, but it's still not great practice to store a password in a world-readable/writable directory like `/tmp`. It's vulnerable to symlink hijacking and unwanted disclosure as-is. Consider putting this inside the archive (which is already protected by containing directory permissions), or better just use `tempfile.mkstemp()` / `with tempfile.TemporaryDirectory() as dir` to create a secure temporary path to store this in. Using the context manager style (`with ...`) has the benefit that you don't have to worry about cleaning up explicitly.

Reply:
This looks good! It removes the potential for `/tmp` file races, eliminates the use of shell/environment variables in the command list and writes the password out before calling any of the kube commands. Thanks for working to add these improvements to the plugin.